mqtt/mqtt-bridge/src/persist/mod.rs

```diff
@@ -0,0 +1,113 @@
+use std::cmp::Ordering;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use futures_util::stream::Stream;
+use mqtt3::proto::Publication;
+use thiserror::Error;
+
+mod memory;
+
+// Queue used in bridge.
+#[async_trait]
+trait Persist<'a> {
+    type Loader: Stream;
+
+    fn new() -> Self;
+
+    async fn insert(
+        &mut self,
+        priority: u32,
+        ttl: Duration,
+        message: Publication,
+    ) -> Result<Key, QueueError>;
+
+    async fn remove(&mut self, key: Key) -> Result<bool, QueueError>;
```

Comment:
nit suggestion on the interface. Why not follow what std collections usually do for remove methods
```
 async fn remove(&mut self, key: Key) -> Option<Publication>;
 ```